js/app/historyStore.js

```diff
@@ -369,6 +381,55 @@ var HistoryStore = new Lang.Class({
         }
         this.set_current_item(HistoryItem.HistoryItem.new_from_object(target_item));
     },
+
+    /**
+     * Method: can_share
+     *
+     * Returns whether the current item can be shared on social networks or not.
+     */
+    can_share: function () {
+        let item = this.get_current_item();
+        return (item !== null &&
+                item.model !== null &&
+                item.model instanceof Eknc.ContentObjectModel &&
+                item.model.original_uri !== null &&
+                item.model.original_uri !== "");
+    },
+
+    /**
+     * Method: share
+     *
+     * Share current item on a social network.
+     */
+    share: function (network) {
+        if (!this.can_share())
+            return;
+        let item = this.get_current_item();
+
+        let original_uri = encodeURIComponent(item.model.original_uri);
+        let uri = null;
+
+        switch(network) {
+            case Network.FACEBOOK:
+                uri = 'https://www.facebook.com/dialog/share?app_id=407909575958642&display=popup&hashtag=%23SharedFromEndless&href='
+                      + original_uri;
```

Comment:
I tested this, and works fine. But can we leave that id publicly like this?

Reply:
Yes, there is no other way to do it, besides its like the public id of "Endless Social" facebook app